qiskit_ibm_runtime/runtime_job.py

```diff
@@ -97,6 +100,7 @@ def __init__(
             Union[Type[ResultDecoder], Sequence[Type[ResultDecoder]]]
         ] = None,
         image: Optional[str] = "",
+        service: "qiskit_runtime_service.QiskitRuntimeService" = None,
```

Comment:
Does `service` need to be optional? If it indeed needs to be optional, then line 268 needs to check if it's None.